src/redux/cognito/cognito.thunk.ts

```diff
@@ -81,6 +81,22 @@ export const signupUserThunk = createAsyncThunk<CognitoState, {
   }
 );
 
+export const updateUserThunk = createAsyncThunk<string, {
+  email: string;
+  preferredUsername?: string;
+}>(
+  'cognito/updateUser',
+  async({ email, preferredUsername }) => {
+    let user = await Auth.currentAuthenticatedUser();
+    const response = await Auth.updateUserAttributes(user, {
+      preferred_username: preferredUsername,
+      email: email
+    });
+
+    return response;
+  }
+);
+
```

Comment:
we are going to have to update some state items. 

Take a look at the items starting at line ~170 or so to get an idea for how to update the state to reflect these new values.